### DISCUSSION TITLE
64bit stringbuilder

### DISCUSSION THREAD

**Discussion by Job-Bouwhuis** [dotnet/runtime#87558] [General]
like either increase the existing one, or add a new class that would be the 64bit version.
i need more than 2 billion characters :/

**Comment by gfoidl** [comment]
>  i need more than 2 billion characters :/

The limitation comes from `System.String` itself, not the `StringBuilder` (as the title alludes).

Why do you need more than 2 billion characters?
Nowadays there are lots of APIs that take Span, etc. so you can work in chunks too.

**Reply by Job-Bouwhuis** [reply]
i want to read a large file into memory, and it seems it has more than 2 billion characters... ill look into using Span. thanks for the answer!

**Reply by gfoidl** [reply]
You can also use "memory mapped files" for this, thus reading it by chunks (via Span, etc).
Reading such a big file once into memory isn't almost always a good idea.

Or you can use `System.IO.Pipelines` for such a task. Pleas have look what this is, then you hopefully see how it can help you.

**Reply by TahirAhmadov** [reply]
Yes, definitely the wrong approach to load a 2GB file into memory. Use a `FileStream` together with `StreamReader` (assuming this is a text file, since the question was about `StringBuilder`) to process the file one chunk at a time.

**Reply by TahirAhmadov** [reply]
Also, `StringBuilder` is a BCL class, it's not a C# language question, so these kinds of questions belong in https://github.com/dotnet/runtime/discussions/

**Reply by huoyaoyuan** [reply]
You can also use `System.IO.RandomAccess` to work with the file.
